addons/mrp/models/mrp_workorder.py

```diff
@@ -619,9 +653,8 @@ def button_start(self, raise_on_invalid_state=False):
                 vals['date_finished'] = leave.date_to
                 vals['leave_id'] = leave.id
                 wo.write(vals)
-            else:
+            elif not wo._has_started():
```

Comment:
Before, the 'progress' state could only transition to 'done' or 'cancel', but as now, it can go back to waiting, ready, ... This check is needed to avoid overwriting already established data